platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentActivationOptions.java

```diff
@@ -0,0 +1,241 @@
+    private LocationEngine locationEngine;
+    private LocationEngineRequest locationEngineRequest;
+    private LocationComponentOptions locationComponentOptions;
+    private int styleRes;
+
+    /**
+     * Set to true as the default in case a true/false value isn't declared via the builder's
+     * {@link LocationComponentActivationOptions#useDefaultLocationEngine()} method.
+     * <p>
+     * Please be aware that this activation boolean is ignored when a non-null
+     * {@link LocationEngine} is set via the builder's `locationEngine()` method.
+     */
+    private boolean useDefaultLocationEngine = true;
+
+    /**
+     * Constructor for the {@link LocationComponentActivationOptions} builder class.
+     * While other activation options are optional, the activation process always requires
+     * context and a fully-loaded map {@link Style} object, which is why the two are in this
+     * constructor.
+     */
+    public Builder(@NonNull Context context, @NonNull Style style) {
+      this.context = context;
+      this.style = style;
+    }
+
+    /**
+     * Deliver your own {@link LocationEngine} to the LocationComponent.
+     * <p>
+     * The true/false
+     * {@link LocationComponentActivationOptions#builder(Context, Style)#useDefaultLocationEngine}
+     * activation option is ignored when a non-null {@link LocationEngine} is set via
+     * this `locationEngine()` method.
+     *
+     * @param locationEngine a {@link LocationEngine} object
+     * @return the {@link Builder} object being constructed
+     */
+    @NonNull
+    public Builder locationEngine(@Nullable LocationEngine locationEngine) {
+      this.locationEngine = locationEngine;
+      return this;
+    }
+
+    /**
+     * @param locationEngineRequest the location request which the
+     *                              {@link LocationComponent} should use
+     * @return the {@link Builder} object being constructed
+     */
+    public Builder locationEngineRequest(LocationEngineRequest locationEngineRequest) {
+      this.locationEngineRequest = locationEngineRequest;
+      return this;
+    }
+
+    /**
+     * @param locationComponentOptions a built {@link LocationComponentOptions} object,
+     *                                 which holds the various {@link LocationComponent}
+     *                                 styling options
+     * @return the {@link Builder} object being constructed
+     */
+    public Builder locationComponentOptions(LocationComponentOptions locationComponentOptions) {
+      this.locationComponentOptions = locationComponentOptions;
+      return this;
+    }
+
+    /**
+     * @param styleRes the LocationComponent style resource (e.g. R.style.style_name)
+     * @return the {@link Builder} object being constructed
+     */
+    public Builder styleRes(int styleRes) {
+      this.styleRes = styleRes;
+      return this;
+    }
+
+    /**
+     * @param useDefaultLocationEngine true if you want to initialize and use the
```

Comment:
Could you add a short note that this is ignored when the non-null engine is set?

Reply:
Ok, now reads as 

```
 /**
     * @param useDefaultLocationEngine true if you want to initialize and use the
     *                                 built-in location engine or false if there
     *                                 should be no location engine initialized
     *                                 This is ignored when null is set as the parameter
     *                                 for {@link LocationComponentActivationOptions#builder
     *                                 (Context, Style)#locationEngine()}.
     * @return the {@link Builder} object being constructed
     */
```